pallets/pallet-bonded-coins/src/tests/transactions/burn_into.rs

```diff
@@ -1 +1,726 @@
+	let pool_id = calculate_pool_id(&[DEFAULT_BONDED_CURRENCY_ID]);
+
+	ExtBuilder::default()
+		.with_pools(vec![(
+			pool_id.clone(),
+			generate_pool_details(
+				vec![DEFAULT_BONDED_CURRENCY_ID],
+				get_linear_bonding_curve(),
+				true,
+				Some(PoolStatus::Refunding),
+				None,
+				Some(DEFAULT_COLLATERAL_CURRENCY_ID),
+				Some(ACCOUNT_00),
+			),
+		)])
+		.build()
+		.execute_with(|| {
+			let origin = RawOrigin::Signed(ACCOUNT_00).into();
+			assert_err!(
+				BondingPallet::burn_into(origin, pool_id, 0, ACCOUNT_00, 1, 2, 1),
+				Error::<Test>::PoolNotLive
+			);
+		});
+}
+
+#[test]
+fn burn_not_hitting_minimum() {
+	let pool_id = calculate_pool_id(&[DEFAULT_BONDED_CURRENCY_ID]);
+
+	ExtBuilder::default()
+		.with_native_balances(vec![(ACCOUNT_00, ONE_HUNDRED_KILT)])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.with_pools(vec![(
+			pool_id.clone(),
+			generate_pool_details(
+				vec![DEFAULT_BONDED_CURRENCY_ID],
+				get_linear_bonding_curve(),
+				true,
+				None,
+				None,
+				Some(DEFAULT_COLLATERAL_CURRENCY_ID),
+				None,
+			),
+		)])
+		.build()
+		.execute_with(|| {
+			let origin = RawOrigin::Signed(ACCOUNT_00).into();
+
+			// burn operation would return less than minimum return
+			assert_err!(
+				BondingPallet::burn_into(origin, pool_id, 0, ACCOUNT_00, 1, 100000, 1),
+				Error::<Test>::Slippage
+			);
+		});
+}
+
+#[test]
+fn burn_invalid_currency_index() {
+	let pool_id = calculate_pool_id(&[DEFAULT_BONDED_CURRENCY_ID]);
+
+	ExtBuilder::default()
+		.with_pools(vec![(
+			pool_id.clone(),
+			generate_pool_details(
+				vec![DEFAULT_BONDED_CURRENCY_ID],
+				get_linear_bonding_curve(),
+				true,
+				None,
+				None,
+				Some(DEFAULT_COLLATERAL_CURRENCY_ID),
+				None,
+			),
+		)])
+		.build()
+		.execute_with(|| {
+			let origin = RawOrigin::Signed(ACCOUNT_00).into();
+
+			// Index beyond array length
+			assert_err!(
+				BondingPallet::burn_into(origin, pool_id, 5, ACCOUNT_00, 1, 2, 1),
+				Error::<Test>::IndexOutOfBounds
+			);
+		});
+}
+
+#[test]
+fn burn_beyond_balance() {
+	let pool_id = calculate_pool_id(&[DEFAULT_BONDED_CURRENCY_ID]);
+
+	ExtBuilder::default()
+		.with_native_balances(vec![(ACCOUNT_00, ONE_HUNDRED_KILT)])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.with_bonded_balance(vec![
+			(DEFAULT_BONDED_CURRENCY_ID, ACCOUNT_00, 1), // Only 1 unit available
+			(DEFAULT_COLLATERAL_CURRENCY_ID, pool_id.clone(), LARGE_BALANCE),
+		])
+		.with_pools(vec![(
+			pool_id.clone(),
+			generate_pool_details(
+				vec![DEFAULT_BONDED_CURRENCY_ID],
+				get_linear_bonding_curve(),
+				true,
+				None,
+				None,
+				Some(DEFAULT_COLLATERAL_CURRENCY_ID),
+				None,
+			),
+		)])
+		.build()
+		.execute_with(|| {
+			let origin = RawOrigin::Signed(ACCOUNT_00).into();
+
+			assert_err!(
+				BondingPallet::burn_into(origin, pool_id, 0, ACCOUNT_00, 2, 0, 1), // Attempt to burn 2 units
+				TokenError::FundsUnavailable
+			);
+		});
+}
```

Comment:
oh yeah that was an issue right?

Reply:
8435ab5e368a89c847b251b1f45135116bdc43e4

Reply:
I think we also need another tx `burn_from` or not? If the pool is permissioned, and the pool creator should delete some funds, there is not really a way right now.

Reply:
Or we change the implementation. If the pool is locked, the bonded currencies are burned from the beneficiary, Otherwise by the caller. But not sure, if it is a good idea to overload the tx.